client/src/elm/Pages/MessagingCenter/Utils.elm

```diff
@@ -108,11 +108,48 @@ surveyQuestionsAnswered surveyType surveyForm =
     Dict.size surveyForm == List.length surveyQuestions
 
 
-resolveSurveyScoreDialogState : ResilienceSurveyType -> Int -> SurveyScoreDialogState
-resolveSurveyScoreDialogState surveyType score =
+resolveSurveyScoreDialogState : NurseId -> ResilienceSurveyType -> Int -> ModelIndexedDb -> SurveyScoreDialogState
+resolveSurveyScoreDialogState nurseId surveyType score db =
     case surveyType of
         ResilienceSurveyQuarterly ->
             QuarterlySurveyScore score
 
         ResilienceSurveyAdoption ->
-            AdoptionSurveyScore score
+            let
+                surveys =
+                    Dict.get nurseId db.resilienceSurveysByNurse
+                        |> Maybe.andThen RemoteData.toMaybe
+                        |> Maybe.map Dict.values
+                        |> Maybe.withDefault []
+
+                adoptionSurveys =
+                    List.filter (\survey -> survey.surveyType == ResilienceSurveyAdoption) surveys
```

Comment:
1. Better syntax:
 `List.filter (.surveyType >> (==) ResilienceSurveyAdoption) surveys`.

2. Lets add this filter as part of `surveys = ...` .